packages/gatsby-remark-prismjs/src/__tests__/load-prism-language-extension.js

```diff
@@ -0,0 +1,130 @@
+const loadLanguageExtension = require(`../load-prism-language-extension`)
+const Prism = require(`prismjs`)
+
+describe(`extend/add prism language`, () => {
+  afterEach(() => {
+    //jest.resetModules()
+  })
+  it(`should throw an error if the request is not an array or an object`, () => {
+    let request = 4
+
+    expect(() => loadLanguageExtension(request)).toThrow(
+      `'languageExtensions' needs to be defined as an array or a JSON object.`
+    )
+
+    request = `A weird string value, instead of an object or array`
+
+    expect(() => loadLanguageExtension(request)).toThrow(
+      `'languageExtensions' needs to be defined as an array or a JSON object.`
+    )
+  })
+  it(`should not throw an error if the request is an object or an array`, () => {
+    let request = []
+
+    expect(() => loadLanguageExtension(request)).not.toThrow()
+
+    request = {}
+
+    expect(() => loadLanguageExtension(request)).not.toThrow()
+  })
+  it(`should extend pre-loaded language`, () => {
+    const request = {
+      extend: `clike`,
+      definition: {
+        flexc_keyword: `(__cm|__circ|_lpp_indirect|__accum|__size_t|__ptrdiff_t|__wchar_t|__fixed|__abscall|__extcall|__stkcall|__sat|__i64_t|__i32_t|__i16_t|__r32_t|__r16_t|__u64_t|__u32_t|__u16_t|__a40_t|__a24_t)`,
+      },
+    }
+
+    loadLanguageExtension(request)
+
+    expect(Prism.languages).toHaveProperty(request.extend)
+    expect(Prism.languages[request.extend]).toHaveProperty(`flexc_keyword`)
+    expect(Prism.languages[request.extend][`flexc_keyword`]).toEqual(
+      request.definition.flexc_keyword
+    )
+  })
+  it(`should extend not pre-loaded language`, () => {
+    const request = {
+      extend: `c`,
+      definition: {
+        flexc_keyword: `(__cm|__circ|_lpp_indirect|__accum|__size_t|__ptrdiff_t|__wchar_t|__fixed|__abscall|__extcall|__stkcall|__sat|__i64_t|__i32_t|__i16_t|__r32_t|__r16_t|__u64_t|__u32_t|__u16_t|__a40_t|__a24_t)`,
+      },
+    }
+
+    loadLanguageExtension(request)
+
+    expect(Prism.languages).toHaveProperty(request.extend)
```

Comment:
This is the same thing as asserting

```
expect(Prism.languages[request.extend]).toBeDefined()
```

right? Is there a reason to go with this slightly less know matcher?